### PR TITLE
ci: automate bumping ZITADEL version

### PR DESCRIPTION
**Pull request by eliobischof**
contributes to #29 

What's happening:
- The action consumes zitadel-released repository_dispatch events. With [this PR on ZITADELs end](https://github.com/zitadel/zitadel/pull/4544), the webhook is called after a ZITADEL release.
- The action compares the newly released ZITADEL version with the appVersion in Charts.yaml
- If the appVersion is patched, the chart version is patched. If the appVersion has a new minor version, the charts minor version is bumped, The same for the major version.
- The action creates or overwrites a pull request, approves the PR and enables auto-merge

I created the PR from a fork for testing purposes, because the workflow is just activated when it's on the default branch.

### CODE REVIEW

File: .github/workflows/bump.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        echo "Chart Version: ${{ steps.current-chart-version.outputs.data }}"
+        echo "ZITADEL Version: ${{ steps.current-zitadel-version.outputs.data }}"
+
+    - name: Parse Last ZITADEL Version
+      id: parsed-last-zitadel-version
+      uses: booxmedialtd/ws-action-parse-semver@v1
+      with:
+        input_string: ${{ steps.current-zitadel-version.outputs.data }}
+
+    - name: Set Version Type
```
Comment (thesephirot) [inline .github/workflows/bump.yml:49]: can this not be handled by semantic? this seems a bit complicated and manual

**Comment by hifabienne** [comment]
@eliobischof @fforootd can we merge this? Or do we need something?

**Comment by eliobischof** [comment]
> @eliobischof @fforootd can we merge this? Or do we need something?

@livio-a why is it WIP again? https://github.com/zitadel/DevOps/issues/10#issue-1604411571

**Comment by livio-a** [comment]
> > @eliobischof @fforootd can we merge this? Or do we need something?
> 
> @livio-a why is it WIP again? [zitadel/DevOps#10 (comment)](https://github.com/zitadel/DevOps/issues/10#issue-1604411571)

Probably only because of the open discussion above (version parsing), but for me it's ok if we merge the current state

File: .github/workflows/bump.yml (hunk 1 above)
Comment (eliobischof) [reply .github/workflows/bump.yml:49]: the semantic release action still doesn't output that information, so we would have to hack eighter way  https://github.com/marketplace/actions/action-for-semantic-release#outputs

Resolving this conversation

**Comment by eliobischof** [comment]
> > > @eliobischof @fforootd can we merge this? Or do we need something?
> > 
> > 
> > @livio-a why is it WIP again? [zitadel/DevOps#10 (comment)](https://github.com/zitadel/DevOps/issues/10#issue-1604411571)
> 
> Probably only because of the open discussion above (version parsing), but for me it's ok if we merge the current state

I resolved it

**Review by livio-a — APPROVED** [review]

**Pull request merged**